client/cypress/integration/dashboard/dashboard_spec.js

```diff
@@ -109,6 +109,154 @@ describe('Dashboard', () => {
     });
   });
 
+  describe('Sharing', () => {
+    beforeEach(function () {
+      createDashboard('Foo Bar').then(({ slug, id }) => {
+        this.dashboardId = id;
+        this.dashboardUrl = `/dashboard/${slug}`;
+      });
+    });
+
+    it('is possible if all queries are safe', function () {
+      const options = {
+        parameters: [{
+          name: 'foo',
+          type: 'number',
+        }],
+      };
+
+      const dashboardUrl = this.dashboardUrl;
+      createQuery({ options }).then(({ id: queryId }) => {
+        cy.visit(dashboardUrl);
+        editDashboard();
+        cy.contains('a', 'Add Widget').click();
+        cy.getByTestId('AddWidgetDialog').within(() => {
+          cy.get(`.query-selector-result[data-test="QueryId${queryId}"]`).click();
+        });
+        cy.contains('button', 'Add to Dashboard').click();
+        cy.getByTestId('AddWidgetDialog').should('not.exist');
+        cy.clickThrough({ button: `
+          Done Editing
+          Publish
+        ` },
+        `OpenShareForm
+        PublicAccessEnabled`);
+
+        cy.getByTestId('SecretAddress').should('exist');
+      });
+    });
+
+    describe('is available to unauthenticated users', () => {
+      const addWidgetAndShareDashboard = (dashboardUrl, query, options, callback) => {
+        createQuery({ query, options }).then(({ id: queryId }) => {
+          cy.visit(dashboardUrl);
+          editDashboard();
+          cy.contains('a', 'Add Widget').click();
+          cy.getByTestId('AddWidgetDialog').within(() => {
+            cy.get(`.query-selector-result[data-test="QueryId${queryId}"]`).click();
+          });
+          cy.contains('button', 'Add to Dashboard').click();
+          cy.getByTestId('AddWidgetDialog').should('not.exist');
+          cy.clickThrough({ button: `
+            Done Editing
+            Publish
+          ` },
+          `OpenShareForm
+          PublicAccessEnabled`);
+
+          cy.getByTestId('SecretAddress').invoke('val').then((secretAddress) => {
+            callback(secretAddress);
+          });
+        });
+      };
+
+      it.only('when there are no parameters', function () {
```

Comment:
@rauchy a mistake?

Reply:
Shouldn't CircleCI frontend-lint be warning those things? :thinking:

Reply:
Nah, [cypress-io/eslint-plugin-cypress](https://github.com/cypress-io/eslint-plugin-cypress) doesn't have that :/

Reply:
Yep. Thanks @gabrieldutra!

Reply:
@rauchy do you wanna enable the other tests?

Reply:
> a mistake?

Totally 🤦‍♂️

> Shouldn't CircleCI frontend-lint be warning those things? 

I remember we had this discussion a while ago :)

Reply:
See #3995